superset/views/base.py

```diff
@@ -63,9 +63,12 @@ def get_error_msg():
 
 
 def json_error_response(msg=None, status=500, stacktrace=None, payload=None, link=None):
+    stack_enabled = conf.get('SHOW_STACKTRACE')
+    if not stack_enabled:
+        msg = 'An error has occurred.'
```

Comment:
If a `msg` is passed we should just use it regardless of `SHOW_STACKTRACE`. `SHOW_STACKTRACE` should only affect the `stacktrace` key in the payload, not the msg itself.